contracts/RewardPool.sol

```diff
@@ -256,6 +289,179 @@ contract RewardPool is AccessControl, ReentrancyGuard {
     emit Withdraw(amount);
   }
 
+  /**
+   * @dev Internal function to get the period id
+   * @param periodStart Start of the reward period (unix timestamp)
+   * @param periodEndExclusive End of the reward period (unix timestamp)
+   * @return bytes32 period id
+   */
+  function _getPeriodId(
+    uint48 periodStart,
+    uint48 periodEndExclusive
+  ) internal pure returns (bytes32) {
+    if (periodStart >= periodEndExclusive)
+      revert PeriodInvalid(periodStart, periodEndExclusive);
+
+    return keccak256(abi.encode(periodStart, periodEndExclusive));
+  }
+
+  /**
+   * @dev Allows the owner to update kpis for a reward period
+   * @param kpis Array of kpis to add
+   * @param periodStart Start of the reward period (unix timestamp)
+   * @param periodEndExclusive End of the reward period (unix timestamp)
+   * @param kpiFunctionId Bytes32 identifier of the kpi function (e.g., github commit hash)
+   * @notice Allowed only for address with DEFAULT_ADMIN_ROLE
+   */
+  function updatePeriodKpis(
+    IRewardFunction.Kpi[] calldata kpis,
+    uint48 periodStart,
+    uint48 periodEndExclusive,
+    bytes32 kpiFunctionId
+  ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    bytes32 periodId = _getPeriodId(periodStart, periodEndExclusive);
+
+    PeriodData storage period = _periods[periodId];
+
+    if (period.processedAt != 0) revert PeriodAlreadyProcessed(periodId);
+
+    for (uint256 i = 0; i < kpis.length; i++) {
+      if (kpis[i].referrerAddress == address(0))
+        revert ZeroAddressNotAllowed(i);
+
+      period.users.add(kpis[i].referrerAddress);
```

Comment:
The PR description states that `0 kpis are treated as no kpis (can be used for deletion)` -- did we decide that we should keep the `referrerAddress` in the user set even if we update that referrer's KPI value to 0? Does it make sense to remove it from the set, if present, if the given KPI value is 0?

Reply:
The description was written before switching to enumerable set / map, where deletes are technically no longer required to keep track of a users list. I didn't like the assumption that `0` kpi means delete. Maybe this is overkill but we could add a `deleteKpi` method to remove